runbooks/mcp-to-permissioned-fps.md

```diff
@@ -0,0 +1,382 @@
+
+## Running a proposer and challenger
+
+You can safely run an instance of `op-proposer` and `op-challenger` after you’ve deployed the contracts for the `op-contracts/v1.6.0` upgrade and validated the upgrade bundle. It is not necessary to shut down your existing `op-proposer` instance. Once the upgrade finalization transaction is executed, the system will seamlessly transition from the `op-contracts/v1.3.0` contracts to the `op-contracts/v1.6.0` contracts.
+
+**Running a proposer**
+
+Refer to the [Releases page](https://github.com/ethereum-optimism/optimism/releases) on the Optimism Monorepo to determine the latest OP Stack release that you should be using.
+
+`op-proposer` has already been updated to support `op-contracts/v1.6.0` if you are using the latest governance-approved version of the OP Stack. You can safely run an instance of `op-proposer` for `op-contracts/v1.6.0` in parallel with your current production instance.
+
+**Configuration**
+
+You can generally use the same configuration that you currently use for `op-proposer` with the following modifications:
+
+- `L2OOAddressFlag`
+    - `--l2oo-address, L2OO_ADDRESS`
+    - Must be empty
+- `DisputeGameFactoryAddressFlag`
+    - `--game-factory-address, GAME_FACTORY_ADDRESS`
+    - Must be set to the address of the `DisputeGameFactoryProxy` contract
+- `DisputeGameTypeFlag`
+    - `--game-type, GAME_TYPE`
+    - Must be set to `1`
+- `ProposalIntervalFlag`
+    - `--proposal-interval, PROPOSAL_INTERVAL`
+    - Recommend setting this to `1h`
+
+**Running a challenger**
+
+Refer to the [Releases page](https://github.com/ethereum-optimism/optimism/releases) on the Optimism Monorepo to determine the latest OP Stack release that you should be using.
+
+> [!NOTE]
+> `op-challenger` v1.1.2 modified the `op-challenger` to disable prestate checks for the permissioned game. This change ensures that the `op-challenger` will correctly resolve games regardless of the configured prestate value.
+
+`op-challenger` is a service that participates in the dispute game process and challenges invalid proposals. The `op-challenger` will NOT have permission to post counterclaims in the permissioned games as the EOA account it uses is not the CHALLENGER role. Under this permissioned setup, the `op-challenger` primarily serves to resolve dispute games on behalf of the proposer and can act as additional monitoring. Refer to [the Optimism Developer Docs](https://docs.optimism.io/builders/chain-operators/tools/op-challenger) for a detailed overview of how to run the `op-challenger`.
+
+**Configuration**
```

Comment:
We also need to set the `trace-type`, `TRACE_TYPE` flag to `permissioned` so the challenger will act on permissioned games.  The default is to act on permissionless (cannon) games.

The flag does take a list so can be set to play both permissioned and permissionless (`--trace-type permissioned,cannon`) but since permissionless games aren't configured there shouldn't be any and we don't wan the challenger posting bonds for them given it won't necessarily have the right config for them.

Reply:
For this case, since we haven't configured cannon games, the challenger will simply fail to start if `cannon` `trace-type` is enabled, that means when you don't specify any `--trace-type` or even include it in the list for types in the args

It fails with error
```
CRIT [10-21|06:53:00.232] Application failed   err="failed to setup: failed to init challenger game service: failed to register game types: failed to register cannon game type: failed to create fault dispute game contracts: failed to retrieve version of dispute game 0x0000000000000000000000000000000000000000: failed to unpack result: failed to unpack data: abi: attempting to unmarshal an empty string while arguments are expected"
```

On start it'll register the game types and will fail when creating a new instance of fault game contract since there is no impl added in factory for cannon game type.

Reply:
That will be fixed in the next release. :) https://github.com/ethereum-optimism/optimism/pull/12539

But yes, since we don't intend to use permissionless games it's better to configure the challenger not to act on them even if they could exist.